### ISSUE TITLE
Add support for annotations and labels

### ISSUE THREAD

**Issue by mikebrow** [containerd/cri#124]
generateSandboxContainerSpec() is currently filtering all annotations received through the CRI api vs copying them to the runtime spec. Consider if some of the annotations should be filtered or if all should be copied over. For example:
```
annotations := config.GetAnnotations()
for key, value := range annotations {
 		g.AddAnnotation(key, value)
}
```
Similarly, labels received over CRI are not being processed / handed over to containerd. 

Consider the patterns enabled/being used for kubernetes when connected to dockershim and CRI-O, should we be unique here?

**Comment by Random-Liu** [comment]
There were some discussion about this before, could not find the email threads now.
The conclusion was that we don't want to expose the container label as part of api, user should not rely on it. :)

**Comment by mikebrow** [comment]
interesting.. what about annotations?

**Comment by mikebrow** [comment]
addressed by #1084
